### ISSUE TITLE
Update BUILD_INSTRUCTIONS.markdown to match latest Docker developments

### ISSUE THREAD

**Issue by dominic-mulligan-arm** [veracruz-project/veracruz#86]
**Describe the bug**
The `BUILD_INSTRUCTIONS.markdown` is hard to follow, especially in light of our recent Docker changes.

**To Reproduce**
Follow the `BUILD_INSTRUCTIONS.markdown` file in conjunction with the contents of `README.md` in `veracruz-docker-image`.

**Expected behaviour**

A few points of inconsistency, or things that could be clarified, stand out:
1. The Intel IAS service does not give you anything that it describes as a "token", yet this is mentioned in the `BUILD_INSTRUCTIONS.markdown` file.  Rather, a primary and a secondary key are generated on your behalf.
2. It's not clear whether you should sign up to a linkable or an unlinkable EPID account from the instructions, or whether this matters.
3. The process for building the Docker container now differs from what is described in the build instructions, and is more accurately reflected in the `README.md` file in `veracruz-docker-image`.

**Comment by dominic-mulligan-arm** [comment]
cc: @geky and @gbryant-arm as you're (relatively) new to the project: if you spot any other inconsistencies in the build instructions then feel free to add them here.

**Comment by geky** [comment]
Since this document changes with the `veracruz-docker-image` repo, does it make since to instead have most of the Docker-related documentation moved into `veracruz-docker-image/README.md`? It would be kind of annoying to have the guide split, but that way it would be more likely to be up to date.

**Comment by dominic-mulligan-arm** [comment]
I think so.  There's also issues in the `README.md` in the Docker repository, too.  That doesn't mention anything about setting the `IAS_TOKEN` or `VERACRUZ_ROOT` environment variables.  I think the material related to that in `BUILD_INSTRUCTIONS.markdown` should be moved there.

**Comment by gbryant-arm** [comment]
The `BUILD_INSTRUCTIONS.markdown` doesn't say we should initialise the `TEE` environment variable (compilation fails if we don't)

**Comment by gbryant-arm** [comment]
* `make sgx-sinaloa-test` should be run inside `veracruz/`
* The instructions to build the TZ image (`make build TEE=tz`) don't show up in `BUILD_INSTRUCTIONS.markdown`

**Comment by gbryant-arm** [comment]
Independently, should we add some instructions for people using a VPN? I overcame several issues with Docker and VPN

**Comment by basmaelgaabouri** [comment]
I think #114 solves this issue.
